### DISCUSSION TITLE
To use Blatann in raspberry?

### DISCUSSION THREAD

**Discussion by ghost** [ThomasGerstenberg/blatann#85] [Q&A] [answered]
Hi Thomas
I know this package is not compatible with raspberry yet,
but  is there any way to use your Blatann in raspberry pi 3? 
Thanks!

**Comment by hientominh** [comment, marked as answer]
Hi,
You can try docker image https://hub.docker.com/repository/docker/vantixinc/raspberry-pi-pc-ble-driver (vantixinc/raspberry-pi-pc-ble-driver:0.14.2-python3.7.9), it already has pc-ble-driver 0.14.2 with python 3.7.
Requirement: docker and docker cli
Step:
- Run docker image as a docker container
- Exec command: python3.7 -m pip install cffi (blatann uses cffi backend)
- Git clone blatann source: git clone https://github.com/ThomasGerstenberg/blatann.git
- Then cd to directory and run: python3.7 setup.py build && python3.7 setup.py install
- Now you can use blatann on raspberry pi

Refer to l[ayer of this image](https://hub.docker.com/layers/vantixinc/raspberry-pi-pc-ble-driver/0.14.2-python3.7.9/images/sha256-4848fd89cb6151447ebb23e6d7470bcd2f8b8fd2ec05a7bac84a2cf9600609de?context=repo) if you want to install this package on python root path raspbian instead of in docker container.

**Comment by ThomasGerstenberg** [comment]
Thanks for the info @hientominh, raspberry pi is one platform I haven't had a chance to test out.
Will leave this issue open for anyone else curious

**Comment by ghost** [comment]
I tried blatann on raspberry pi and installed it. 

@ThomasGerstenberg  However the Bonding information is stored in the *package installation path* which is not writeable by default.

 ```/usr/local/lib/python3.7/dist-packages/blatann-0.3.4-py3.7.egg/blatann/.user/bonding_db.pkl```

Seems like this default path cannot be changed when using 
```class BleDevice(NrfDriverObserver):```

I would propose to store this in a more standard location like `$HOME/.blatann` or `$HOME/.config/blatann` instead.
Would you like to get a patch for this ?

**Comment by ThomasGerstenberg** [comment]
Thanks for the feedback @OlafFlebbeBosch, I've created #60 to track that. I've been also meaning to move away from using pickle to store bonding info, will look into that for the next release as well.

In the meantime, the bond database loader can be swapped out before opening and point to a different file

```python
import os
from blatann.gap.default_bond_db import DefaultBondDatabaseLoader

bond_db_file = os.path.join(os.path.expanduser("~"), ".blatann", "bonding_db.pkl")

device = BleDevice(comport)
device.bond_db_loader = DefaultBondDatabaseLoader(bond_db_file)
device.open()
```
